ebpf_ffi/ebpf.cc

```diff
@@ -21,7 +21,27 @@ namespace ebpf_ffi {
 constexpr size_t kLogBuffSize = 100000000;
 }  // namespace ebpf_ffi
 
-int load_ebpf_program(void *prog_buff, size_t prog_size,
+int btf_load(void *btf_buff, size_t btf_size, std::string *error) {
+  union bpf_attr btf_attr;
+  memset(&btf_attr, 0, sizeof(btf_attr));
+  btf_attr.btf = (uint64_t)btf_buff;
+  btf_attr.btf_size = btf_size;
+
+  char *btf_log_buf = (char *)malloc(1024);
+  memset(btf_log_buf, 0, 1024);
+  btf_attr.btf_log_buf = (uint64_t)btf_log_buf;
+  btf_attr.btf_log_size = 1024;
+  btf_attr.btf_log_level = 2;
+
+  int btf_fd = syscall(SYS_bpf, BPF_BTF_LOAD, &btf_attr, sizeof(btf_attr));
```

Comment:
My guess is that it will return -1. To prevent prog load from failing we could check if the btf_fd is valid before attaching it in ` attr.prog_btf_fd = btf_fd;`..?

Reply:
that could work yes :)

Reply:
Done